### PR TITLE
Implement `selected_char_kind` and remove lfortran_intrinsic_kind.f90

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: https://github.com/lfortran/lfortran/issues/3488
Continuing: https://github.com/lfortran/lfortran/pull/3518

### CODE REVIEW

File: a.f90 [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    integer, parameter :: default  = selected_char_kind ("default")
+    integer, parameter :: ucs4  = selected_char_kind ("ISO_10646")
+    integer, parameter :: other = selected_char_kind ("other")
+    character(5) :: ascii_str = "ascii"
+    character(5) :: ascii_case_str = "ASciI"
+    character(7) :: default_str = "default"
+    character(9) :: ucs4_str = "ISO_10646"
+    character(5) :: other_str = "other"
+
+    print*, selected_char_kind(ascii_str)
```
Comment (HarshitaKalani) [inline a.f90:14]: This example throws the following error:
```console
$ lfortran a.f90
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in _start()
  File "./csu/../csu/libc-start.c", line 392, in __libc_start_main_impl()
  File "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main()
  File "/home/harshita/Desktop/lfortran/src/bin/lfortran.cpp", line 2394, in ??
    return main_app(argc, argv);
  File "/home/harshita/Desktop/lfortran/src/bin/lfortran.cpp", line 2355, in main_app(int, char**)
    err = compile_to_object_file(arg_file, tmp_o, false,
  File "/home/harshita/Desktop/lfortran/src/bin/lfortran.cpp", line 949, in ??
    res = fe.get_llvm3(*asr, lpm, diagnostics, infile);
  File "/home/harshita/Desktop/lfortran/src/lfortran/fortran_evaluator.cpp", line 360, in LCompilers::FortranEvaluator::get_llvm3(LCompilers::ASR::TranslationUnit_t&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    compiler_options, run_fn, infile);
  File "/home/harshita/Desktop/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 9739, in LCompilers::asr_to_llvm(LCompilers::ASR::TranslationUnit_t&, LCompilers::diag::Diagnostics&, llvm::LLVMContext&, Allocator&, LCompilers::PassManager&, LCompilers::CompilerOptions&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    pass_manager.apply_passes(al, &asr, co.po, diagnostics);
  File "/home/harshita/Desktop/lfortran/src/libasr/pass/pass_manager.h", line 306, in LCompilers::PassManager::apply_passes(Allocator&, LCompilers::ASR::TranslationUnit_t*, LCompilers::PassOptions&, LCompilers::diag::Diagnostics&)
    apply_passes(al, asr, _passes, pass_options, diagnostics);
  File "/home/harshita/Desktop/lfortran/src/libasr/pass/pass_manager.h", line 164, in LCompilers::PassManager::apply_passes(Allocator&, LCompilers::ASR::TranslationUnit_t*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, LCompilers::PassOptions&, LCompilers::diag::Diagnostics&)
    _passes_db[passes[i]](al, *asr, pass_options);
  File "/home/harshita/Desktop/lfortran/src/libasr/pass/intrinsic_function.cpp", line 378, in LCompilers::pass_replace_intrinsic_function(Allocator&, LCompilers::ASR::TranslationUnit_t&, LCompilers::PassOptions const&)
    v.visit_TranslationUnit(unit);
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 46613, in LCompilers::ASR::CallReplacerOnExpressionsVisitor<LCompilers::ReplaceIntrinsicFunctionsVisitor>::visit_TranslationUnit(LCompilers::ASR::TranslationUnit_t const&)
    this->visit_symbol(*a.second);
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 5139, in LCompilers::ASR::BaseVisitor<LCompilers::ReplaceIntrinsicFunctionsVisitor>::visit_symbol(LCompilers::ASR::symbol_t const&)
    void visit_symbol(const symbol_t &b) { visit_symbol_t(b, self()); }
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 4850, in ??
    case symbolType::Program: { v.visit_Program((const Program_t &)x); return; }
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 46624, in LCompilers::ASR::CallReplacerOnExpressionsVisitor<LCompilers::ReplaceIntrinsicFunctionsVisitor>::visit_Program(LCompilers::ASR::Program_t const&)
    self().transform_stmts(xx.m_body, xx.n_body);
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 46606, in LCompilers::ASR::CallReplacerOnExpressionsVisitor<LCompilers::ReplaceIntrinsicFunctionsVisitor>::transform_stmts(LCompilers::ASR::stmt_t**&, unsigned long&)
    self().visit_stmt(*m_body[i]);
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 5156, in LCompilers::ASR::BaseVisitor<LCompilers::ReplaceIntrinsicFunctionsVisitor>::visit_stmt(LCompilers::ASR::stmt_t const&)
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 4890, in ??
    case stmtType::Print: { v.visit_Print((const Print_t &)x); return; }
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 46946, in LCompilers::ASR::CallReplacerOnExpressionsVisitor<LCompilers::ReplaceIntrinsicFunctionsVisitor>::visit_Print(LCompilers::ASR::Print_t const&)
    self().call_replacer();
  File "/home/harshita/Desktop/lfortran/src/libasr/pass/intrinsic_function.cpp", line 146, in LCompilers::ReplaceIntrinsicFunctionsVisitor::call_replacer()
    replacer.replace_expr(*current_expr);
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/asr.h", line 45583, in LCompilers::ASR::BaseExprReplacer<LCompilers::ReplaceIntrinsicFunctions>::replace_expr(LCompilers::ASR::expr_t*)
    self().replace_IntrinsicElementalFunction(down_cast<ASR::IntrinsicElementalFunction_t>(x));
  File "/home/harshita/Desktop/lfortran/src/libasr/pass/intrinsic_function.cpp", line 76, in LCompilers::ReplaceIntrinsicFunctions::replace_IntrinsicElementalFunction(LCompilers::ASR::IntrinsicElementalFunction_t*)
    ASR::expr_t* current_expr_ = instantiate_function(al, x->base.base.loc,
  File "/home/harshita/Desktop/lfortran/src/libasr/../libasr/pass/intrinsic_functions.h", line 2996, in ??
    scope->add_symbol(fn_name, f_sym);
AssertFailed: scope.find(name) == scope.end()
```

Comment (HarshitaKalani) [reply a.f90]: Thanks @Pranavchiku for the help!

File: src/libasr/pass/intrinsic_functions.h [hunk 2]
```diff
@@ -3000,6 +3065,55 @@ namespace SelectedRealKind {
...
 
+namespace SelectedCharKind {
+
+    static inline ASR::expr_t *eval_SelectedCharKind(Allocator &al, const Location &loc,
+            ASR::ttype_t* /*t1*/, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        int64_t kind;
+        ASRUtils::ASRBuilder b(al, loc);
+        char* name = ASR::down_cast<ASR::StringConstant_t>(args[0])->m_s;
+
+        if (strcasecmp(name, "ASCII") == 0 || strcasecmp(name, "DEFAULT") == 0) {
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:3076]: The `strcasecmp` fails on Windows. Let's add a function into utils for this operation and call a proper function on all platforms there. We might already have one.

Comment (certik) [reply src/libasr/pass/intrinsic_functions.h:3076]: Let's name it `case_compare` and inside call `tolower`.

**Review by certik — APPROVED** [review]
Can be merged after the simple fix above.

**Pull request merged**